x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/state_management/stream_enrichment_state_machine/utils.ts

```diff
@@ -101,7 +101,7 @@ export function getStepsForSimulation({ stepRefs }: Pick<StreamEnrichmentContext
 
   // Find if any processor is currently being edited
   const editingProcessorIndex = newStepSnapshots.findIndex(
-    (snapshot) => isActionBlock(snapshot.context) && isStepUnderEdit(snapshot)
+    (snapshot) => isActionBlock(snapshot.context.step) && isStepUnderEdit(snapshot)
```

Comment:
This is the point where the processor under edit was not correctly identified, as the step was never passed into the guard function.